server/services/store/sqlstore/migrations/migrations_files/000015_add_teams_and_boards.up.sql

```diff
@@ -0,0 +1,361 @@
+        ON P.parent_id=GP.id
+INNER JOIN {{.prefix}}blocks_history as GPP
+        ON GP.parent_id=GPP.id
+       SET B.board_id=GPP.id
+     WHERE GPP.type = 'board';
+{{end}}
+{{if .postgres}}
+  UPDATE {{.prefix}}blocks as B
+     SET board_id=P.id
+    FROM {{.prefix}}blocks as P
+   WHERE B.parent_id=P.id
+     AND P.type = 'board';
+
+  UPDATE {{.prefix}}blocks as B
+     SET board_id=GP.id
+    FROM {{.prefix}}blocks as P,
+         {{.prefix}}blocks as GP
+   WHERE B.parent_id=P.id
+     AND P.parent_id=GP.id
+     AND GP.type = 'board';
+
+  UPDATE {{.prefix}}blocks as B
+     SET board_id=GGP.id
+    FROM {{.prefix}}blocks as P,
+         {{.prefix}}blocks as GP,
+         {{.prefix}}blocks as GGP
+   WHERE B.parent_id=P.id
+     AND P.parent_id=GP.id
+     AND GP.parent_id=GGP.id
+     AND GGP.type = 'board';
+
+  UPDATE {{.prefix}}blocks_history as B
+     SET board_id=P.id
+    FROM {{.prefix}}blocks_history as P
+   WHERE B.parent_id=P.id
+     AND P.type = 'board';
+
+  UPDATE {{.prefix}}blocks_history as B
+     SET board_id=GP.id
+    FROM {{.prefix}}blocks_history as P,
+         {{.prefix}}blocks_history as GP
+   WHERE B.parent_id=P.id
+     AND P.parent_id=GP.id
+     AND GP.type = 'board';
+
+  UPDATE {{.prefix}}blocks_history as B
+     SET board_id=GGP.id
+    FROM {{.prefix}}blocks_history as P,
+         {{.prefix}}blocks_history as GP,
+         {{.prefix}}blocks_history as GGP
+   WHERE B.parent_id=P.id
+     AND P.parent_id=GP.id
+     AND GP.parent_id=GGP.id
+     AND GGP.type = 'board';
+{{end}}
+
+
+{{- /* Remove boards that are not templates */ -}}
+{{if .postgres}}
+DELETE FROM {{.prefix}}blocks WHERE type = 'board' AND NOT (fields->'isTemplate')::text::boolean;
+DELETE FROM {{.prefix}}blocks_history WHERE type = 'board' AND NOT (fields->'isTemplate')::text::boolean;
+{{end}}
+{{if .mysql}}
+DELETE FROM {{.prefix}}blocks WHERE type = 'board' AND NOT fields->'$.isTemplate';
+DELETE FROM {{.prefix}}blocks_history WHERE type = 'board' AND NOT fields->'$.isTemplate';
+{{end}}
+{{if .sqlite}}
+DELETE FROM {{.prefix}}blocks WHERE type = 'board' AND NOT json_extract(fields, '$.isTemplate');
+DELETE FROM {{.prefix}}blocks_history WHERE type = 'board' AND NOT json_extract(fields, '$.isTemplate');
+{{end}}
+
+
+{{- /* add board_members */ -}}
+CREATE TABLE {{.prefix}}board_members (
+    board_id VARCHAR(36) NOT NULL,
+    user_id VARCHAR(36) NOT NULL,
+    roles VARCHAR(64),
+    scheme_admin BOOLEAN,
+    scheme_editor BOOLEAN,
+    scheme_commenter BOOLEAN,
+    scheme_viewer BOOLEAN,
+    PRIMARY KEY (board_id, user_id)
+) {{if .mysql}}DEFAULT CHARACTER SET utf8mb4{{end}};
+
+CREATE INDEX idx_boardmembers_user_id ON {{.prefix}}board_members(user_id);
+
+{{if .plugin}}
+{{- /* if we're in plugin, migrate channel memberships to the board */ -}}
+INSERT INTO {{.prefix}}board_members (
+    SELECT B.Id, CM.UserId, CM.Roles, (CM.UserId=B.created_by) OR CM.SchemeAdmin, CM.SchemeUser, FALSE, CM.SchemeGuest
```

Comment:
This is assigning "Viewer" role to guests. I don't think that is wrong, but looks like something that we need to support ASAP in the interface, then.